File: tests/bin/uname.c
// SPDX-License-Identifier: Apache-2.0

#include "libc.h"

int strcmp(const char* s1, const char* s2)
{
    while(*s1 && (*s1==*s2))
        s1++,s2++;
    return *(const unsigned char*)s1-*(const unsigned char*)s2;
}

int main(void) {

    int v;
    char *l = "Linux";
    struct utsname buffer;

    errno = 0;
    if (uname(&buffer) != 0) {
       return 1;
    }

    v = strcmp(buffer.sysname, l);
    if (v != 0) {
        return 1;
    }

    return 0;
}

File: tests/crates/enarx_syscall_tests/src/bin/uname.rs
// SPDX-License-Identifier: Apache-2.0

#![no_std]
#![no_main]
#![feature(asm_sym)]

enarx_syscall_tests::startup!();

use enarx_syscall_tests::*;

fn main() -> Result<()> {
    const LINUX: [i8; 5] = ['L' as i8, 'i' as i8, 'n' as i8, 'u' as i8, 'x' as i8];
    let mut buf = libc::utsname {
        sysname: [0; 65],
        nodename: [0; 65],
        release: [0; 65],
        version: [0; 65],
        machine: [0; 65],
        domainname: [0; 65],
    };

    uname(&mut buf as _)?;

    if !buf.sysname.starts_with(&LINUX) {
        Err(1)
    } else {
        Ok(())
    }
}
